spec/lib/msf/ui/console/command_dispatcher/core_spec.rb

```diff
@@ -397,4 +397,295 @@ def set_tabs_test(option)
       end
     end
   end
+
+  describe "#cmd_sessions" do
+    before(:each) do
+      allow(driver).to receive(:active_session=)
+      allow(framework).to receive(:sessions).and_return(sessions)
+      Timecop.freeze(Time.parse('Dec 18, 2022 12:33:40.000000000 GMT'))
+    end
+
+    context "with no sessions" do
+      let(:sessions) {[]}
+      it "should show an empty table with no args" do
+        core.cmd_sessions
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          Active sessions
+          ===============
+          
+          No active sessions.
+        TABLE
+      end
+
+      it "should show no matches with a search term" do
+        core.cmd_sessions("--search", "session_id:1")
+        expect(@combined_output.join("\n")).to match_table <<~TABLE
+          No matching sessions.
+        TABLE
+      end
+    end
+
+    context "with sessions" do
+      let(:sessions) do 
+        {
+          1 => instance_double(::Msf::Sessions::Meterpreter_x64_Win, last_checkin: Time.now, type: 'meterpreter', sid:1, sname: "sesh1", info: "info", session_host: "127.0.0.1", tunnel_to_s: "tunnel")
```

Comment:
I agree it would be easier to read, I put it this way since we *don't* want them to exist for the "no active sessions" context which exists at the same level as all the other contexts. But if we're fine separating them, or there's another rspec trick I'm not privy to, happy to change that up

Reply:
Additionally, I suppose using the same sessions might be an issue when testing sessions that don't respond to last_checkin, if we want to see what a query returns when none of the sessions do